src/shell/binary/readln.rs

```diff
@@ -166,3 +166,23 @@ fn complete_as_file(current_dir: PathBuf, filename: String, index: usize) -> boo
     // By default assume its not a file
     false
 }
+
+/// prints prompt info lines and
+/// returns the last prompt line.
+fn handle_prompt(full_prompt: String) -> Result<String, String> {
+    if !full_prompt.contains("\n") {
+        return Ok(full_prompt)
+    }
+
+    let stdout = io::stdout();
+    let mut handle = stdout.lock();
+    if let Err(why) =
+        handle.write(full_prompt.as_bytes()) {
+        return Err(format!("unable to print prompt info: {}", why));
+    }
+
+    match full_prompt.lines().last() {
```

Comment:
Try using `rfind('\n')` instead of `lines()` to get the position of the newline for the last line, if you just want to grab the last line without allocating.